### PR TITLE
Specify key names generated for keys with masks

### LINKED ISSUE #1138: Key names for keys with masks
https://github.com/p4lang/p4c/issues/3514 has identified a mismatch between the compiler and the spec: the spec does not allow keys of the form (a & const) without a `@name` annotation, while the compiler does. We should probably fix the spec.

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
Signed-off-by: Mihai Budiu <mbudiu@vmware.com>
Fixes #1138 
This has been supported by the compiler for at least 5 years.

### CODE REVIEW

**Comment by apinski-cavium** [comment]
I assume you double checked that there was no other expressions missing from that table the compiler supports too?
Note I am only getting to the table and control plane parts of the specifications for my compiler so I am happy for this change, I just want to make sure I don't miss any others here.

**Comment by mbudiu-vmw** [comment]
@vgurevich will post comments.

File: p4-16/spec/P4-16-spec.mdk [hunk 1]
```diff
@@ -7683,6 +7683,7 @@ syntactic names:
 | Constants. | `1` | `"1"` |
 | Field projections. | `data.f1` | `"data.f1"` |
 | Slices. | `f1[3:0]` | `"f1[3:0]"` |
+| Masks. | `h.src & 0xFFFF` | `"h.sec & 0xFFFF"` |
```
Comment (mbudiu-vmw) [inline p4-16/spec/P4-16-spec.mdk:7686]: typo in the generated field name

**Comment by mbudiu-vmw** [comment]
The Barefoot compiler does not use these rules for masks and slices.

File: p4-16/spec/P4-16-spec.mdk (hunk 1 above)
Comment (mbudiu-vmw) [reply p4-16/spec/P4-16-spec.mdk:7686]: we could make these just suggestions, and not mandatory rules for all implementations.

File: p4-16/spec/P4-16-spec.mdk [hunk 2]
```diff
@@ -7736,20 +7751,9 @@ table t {
...
-| The `isValid()` method. | `h.isValid()` | `"h.isValid()"` |
-| Array accesses. | `header_stack[1]` | `"header_stack[1]"` |
-| Constants. | `1` | `"1"` |
-| Field projections. | `data.f1` | `"data.f1"` |
-| Slices. | `f1[3:0]` | `"f1[3:0]"` |
-
-All other kinds of expressions **must** be annotated with a `@name`
-annotation (Section [#sec-control-plane-api-annotations]), as in the
-following example.
+If a compiler cannot generate a name for a key is **requires** the key
```
Comment (jafingerhut) [inline p4-16/spec/P4-16-spec.mdk:7754]: is -> it?

**Review by jafingerhut — APPROVED** [review]
Looks reasonable to me.

**Review by jafingerhut — APPROVED** [review]
LGTM

**Pull request merged**